desktop/apps/auction2/components/page/Footer.js

```diff
@@ -2,14 +2,19 @@ import PropTypes from 'prop-types'
 import React from 'react'
 import block from 'bem-cn'
 import renderTemplate from 'desktop/components/react/utils/render_template'
+import { connect } from 'react-redux'
 import { first } from 'underscore'
 
-export default function Footer (props) {
-  const { articles, auction, footerItems } = props
-  const footerItem = first(footerItems)
-  const showArticles = Boolean(articles.length)
-  const showFooterItems = footerItem && !auction.isAuctionPromo()
-  const b = block('auction-footer')
+function Footer (props) {
+  const {
+    articles,
+    footerItem,
+    showArticles,
+    showFooterItems,
+    sd
+  } = props
```

Comment:
@sweir27, @erikdstock - this is a good pattern to know when using redux or any selector-based library -- Perform all logic in selectors (scroll down a bit to bottom of file) so that what the component ends up receiving is simply display values. Then, when performance becomes a concern, its easy to [memoize the selector](https://github.com/reactjs/reselect) so that the component only renders whenever there's a change to the data. (Regarding memoization: This is similar to `shouldComponentUpdate` except that it applies for derived values computed in the selector.)